app/controllers/restaurants_controller.rb

```diff
@@ -11,4 +11,24 @@ def index
   def show
     @restaurant = Restaurant.find(params[:id])
   end
+
+  def new
+  end
+
+  def create
+    @restaurant = Restaurant.new(restaurant_params)
+    if @restaurant.save
+      redirect_to root_path
+      flash[:notice] = "Your restaurant has been sucessfully registered!"
+    else
+      flash[:error] = "Ooops!! something is wrong"
```

Comment:
You know that you have access to `@restaurant.errors` that would allow you to display more detailed message on error than the, in my opinion, quite generic "Ooops!! something is wrong", right?